### PR TITLE
Add deploy section into .travis.yml

### LINKED ISSUE #92: Release process does not work
```shell
$ python setup.py sdist upload -r pypi

running sdist
running egg_info
writing requirements to opentracing_instrumentation.egg-info/requires.txt
writing opentracing_instrumentation.egg-info/PKG-INFO
writing top-level names to opentracing_instrumentation.egg-info/top_level.txt
writing dependency_links to opentracing_instrumentation.egg-info/dependency_links.txt
reading manifest file 'opentracing_instrumentation.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'opentracing_instrumentation.egg-info/SOURCES.txt'
running check
creating opentracing_instrumentation-3.1.0
creating opentracing_instrumentation-3.1.0/opentracing_instrumentation
creating opentracing_instrumentation-3.1.0/opentracing_instrumentation.egg-info
creating opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying files to opentracing_instrumentation-3.1.0...
copying LICENSE -> opentracing_instrumentation-3.1.0
copying MANIFEST.in -> opentracing_instrumentation-3.1.0
copying README.md -> opentracing_instrumentation-3.1.0
copying setup.cfg -> opentracing_instrumentation-3.1.0
copying setup.py -> opentracing_instrumentation-3.1.0
copying opentracing_instrumentation/__init__.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation
copying opentracing_instrumentation/config.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation
copying opentracing_instrumentation/http_client.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation
copying opentracing_instrumentation/http_server.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation
copying opentracing_instrumentation/interceptors.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation
copying opentracing_instrumentation/local_span.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation
copying opentracing_instrumentation/request_context.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation
copying opentracing_instrumentation/utils.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation
copying opentracing_instrumentation.egg-info/PKG-INFO -> opentracing_instrumentation-3.1.0/opentracing_instrumentation.egg-info
copying opentracing_instrumentation.egg-info/SOURCES.txt -> opentracing_instrumentation-3.1.0/opentracing_instrumentation.egg-info
copying opentracing_instrumentation.egg-info/dependency_links.txt -> opentracing_instrumentation-3.1.0/opentracing_instrumentation.egg-info
copying opentracing_instrumentation.egg-info/not-zip-safe -> opentracing_instrumentation-3.1.0/opentracing_instrumentation.egg-info
copying opentracing_instrumentation.egg-info/requires.txt -> opentracing_instrumentation-3.1.0/opentracing_instrumentation.egg-info
copying opentracing_instrumentation.egg-info/top_level.txt -> opentracing_instrumentation-3.1.0/opentracing_instrumentation.egg-info
copying opentracing_instrumentation/client_hooks/__init__.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/_current_span.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/_dbapi2.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/_patcher.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/_singleton.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/boto3.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/celery.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/mysqldb.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/psycopg2.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/requests.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/sqlalchemy.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/strict_redis.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/tornado_http.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/urllib.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
copying opentracing_instrumentation/client_hooks/urllib2.py -> opentracing_instrumentation-3.1.0/opentracing_instrumentation/client_hooks
Writing opentracing_instrumentation-3.1.0/setup.cfg
Creating tar archive
removing 'opentracing_instrumentation-3.1.0' (and everything under it)
running upload
WARNING: Uploading via this command is deprecated, use twine to upload instead (https://pypi.org/p/twine/)
Traceback (most recent call last):
  File "setup.py", line 67, in <module>
    'sphinx_rtd_theme',
  File "/Users/yurishkuro/Uber/opentracing-python-instrumentation/env/lib/python2.7/site-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/usr/local/Cellar/python@2/2.7.15_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/local/Cellar/python@2/2.7.15_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/local/Cellar/python@2/2.7.15_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/Users/yurishkuro/Uber/opentracing-python-instrumentation/env/lib/python2.7/site-packages/setuptools/command/upload.py", line 26, in run
    orig.upload.run(self)
  File "/usr/local/Cellar/python@2/2.7.15_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/command/upload.py", line 62, in run
    self.upload_file(command, pyversion, filename)
  File "/Users/yurishkuro/Uber/opentracing-python-instrumentation/env/lib/python2.7/site-packages/setuptools/command/upload.py", line 133, in upload_file
    value = str(value).encode('utf-8')
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 664: ordinal not in range(128)
```

### PR DESCRIPTION
**Pull request by Jamim**
Hello @yurishkuro,

The main purpose of this PR is to simplify the deployment flow.
Please note that user and encrypted password must be set properly.

Useful links:
 - https://docs.travis-ci.com/user/encryption-keys
 - https://docs.travis-ci.com/user/deployment/pypi
 - https://docs.travis-ci.com/user/deployment#examples-of-conditional-deployment

Fixes #92

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -49,3 +49,13 @@ script:
...
+
+deploy:
+  provider: pypi
+  distributions: 'sdist bdist_wheel'
+  user: 'Your username'
+  password:
+    secure: 'Your encrypted password'
+  on:
+    tags: true
+    python: '3.7'
```
Comment (yurishkuro) [inline .travis.yml]: probably also want to add master branch only

Comment (Jamim) [reply .travis.yml]: It's not possible to use both `tags` and `branch` conditions at the same time.
The `branch` condition will be ignored in this case.

https://docs.travis-ci.com/user/deployment#conditional-releases-with-on
> `tags: true`: deployment is triggered if and only if `$TRAVIS_TAG` is set. Depending on your workflow, you may set `$TRAVIS_TAG` explicitly, even if this is a non-tag build when it was initiated. This causes the `branch` condition to be ignored.

**Comment by Jamim** [comment]
Hello @yurishkuro,
Can you please work on this?
Thanks!

**Comment by Jamim** [comment]
ping @yurishkuro

**Comment by yurishkuro** [comment]
I will add encrypted password separately

**Comment by yurishkuro** [comment]
https://github.com/uber-common/opentracing-python-instrumentation/commit/b5ed8edb5718c9e93451f21adbb0dbe543c9ac81

**Pull request merged**